state-surgery/db.go

```diff
@@ -0,0 +1,27 @@
+package state_surgery
+
+import (
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/ethdb"
+	"github.com/ethereum/go-ethereum/log"
+	"path/filepath"
+)
+
+// MustOpenDB opens a Geth database, or panics. Note that
+// the database must be opened with a freezer in order to
+// properly read historical data.
+func MustOpenDB(dataDir string) ethdb.Database {
+	dir := filepath.Join(dataDir, "geth", "chaindata")
+	db, err := rawdb.NewLevelDBDatabaseWithFreezer(
+		dir,
+		0,
```

Comment:
Does increasing the cache here improve performance?

Reply:
I haven't benchmarked this. It may in theory, so I've added CLI flags to tune this.